test/test.googleauth.ts

```diff
@@ -1252,6 +1252,66 @@ describe('._checkIsGCE', () => {
     assert.equal(false, auth.isGCE);
   });
 
+  it('should retry the check for isGCE if it fails the first time',
+     async () => {
+       const auth = new GoogleAuth();
+       assert.notEqual(true, auth.isGCE);
+       // the first request will fail
+       const scope1 =
+           nock('http://metadata.google.internal').get('/').reply(500);
+       // the second one will succeed
+       const scope2 =
+           nock('http://metadata.google.internal').get('/').reply(200, null, {
+             'metadata-flavor': 'Google'
+           });
+       const isGCE = await auth._checkIsGCE();
+       assert.equal(true, auth.isGCE);
+       assert(scope1.isDone());
+       assert(scope2.isDone());
+     });
+
+  it('should not retry the check for isGCE if it fails with a 404',
+     async () => {
+       const auth = new GoogleAuth();
+       assert.notEqual(true, auth.isGCE);
+       // the first request will fail
+       const scope1 =
+           nock('http://metadata.google.internal').get('/').reply(404);
+       // the second one should never happen
+       const scope2 =
+           nock('http://metadata.google.internal').get('/').reply(404);
+       try {
+         const isGCE = await auth._checkIsGCE();
+         assert.notEqual(true, auth.isGCE);
+       } catch (e) {
+         return;
+       }
+       assert(scope1.isDone());
+       assert(!scope2.isDone());
+       assert.fail('Expected to throw');
+     });
+
+  it('should not retry the check for isGCE if it fails with an ETIMEDOUT',
```

Comment:
The test doesn't match the title. The test seems to be for `ENOTFOUND`. It would be nice to add another test for `ETIMEDOUT` though 😄 .

Reply:
Lol, no idea why ETIMEDOUT came off the fingers.  Fixed.